refactor(transformer/class-properties): move transform logic of `callee` of `CallExpression` to `transform_private_field_callee`

Dunqing opened the pull request:
This transform logic is also can be used to `tag` of `TaggedTemplateExpression`, so let's move it to a separate function.

Dunqing commented:
* **#7504** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7504?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#7503** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7503?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7503?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

overlookmotel reviewed (approved):
I've pushed a commit to include output for static props in comments.

@Dunqing If you're happy with that, please merge.

overlookmotel commented:
I caused merge conflicts with one of my other PRs. I'll fix and merge.

Dunqing committed:
refactor(transformer/class-properties): move transform logic of `callee` of `CallExpression` to `transform_private_field_callee` (#7503)

This transform logic is also can be used to `tag` of `TaggedTemplateExpression`, so let's move it to a separate function.
